plugins/flytekit-huggingface/flytekitplugins/huggingface/sd_transformers.py

```diff
@@ -0,0 +1,71 @@
+import typing
+
+import datasets
+
+from flytekit import FlyteContext
+from flytekit.models import literals
+from flytekit.models.literals import StructuredDatasetMetadata
+from flytekit.models.types import StructuredDatasetType
+from flytekit.types.structured.structured_dataset import (
+    GCS,
+    LOCAL,
+    PARQUET,
+    S3,
+    StructuredDataset,
+    StructuredDatasetDecoder,
+    StructuredDatasetEncoder,
+    StructuredDatasetTransformerEngine,
+)
+
+
+class HuggingFaceDatasetToParquetEncodingHandler(StructuredDatasetEncoder):
+    def __init__(self, protocol: str):
+        super().__init__(datasets.Dataset, protocol, PARQUET)
+
+    def encode(
+        self,
+        ctx: FlyteContext,
+        structured_dataset: StructuredDataset,
+        structured_dataset_type: StructuredDatasetType,
+    ) -> literals.StructuredDataset:
+        df = typing.cast(datasets.Dataset, structured_dataset.dataframe)
+
+        local_dir = ctx.file_access.get_random_local_directory()
+        local_path = f"{local_dir}/00000"
```

Comment:
@wild-endeavor / @pingsutw is it possible to directly stream write instead of write a file and upload directory. I thought we had done this?

Reply:
Ping